unit_test/test_spdm_callback/spdm_cert_verify_callback.c

```diff
@@ -0,0 +1,253 @@
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_vendor)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_vendor, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*model*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 1);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_model)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_model, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*version*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 2);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_version)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_version, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*svn*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 3);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_svn)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_svn, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*layer*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 4);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_layer)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_layer, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*index*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 5);
+    if (result) {
+        ptr += obj_len;
+    }
+    /*fwids*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  (LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC |
+                                   LIBSPDM_CRYPTO_ASN1_CONSTRUCTED) + 6);
+    if (result) {
+        if ((obj_len != sizeof(m_libspdm_dice_tcbinfo_fwids)) ||
+            (!libspdm_consttime_is_mem_equal(ptr, m_libspdm_dice_tcbinfo_fwids, obj_len))) {
+            return false;
+        }
+        ptr += obj_len;
+    }
+    /*flags*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 7);
+    if (result) {
+        ptr += obj_len;
+    }
+    /*vendorInfo*/
+    result = libspdm_asn1_get_tag(&ptr, end, &obj_len,
+                                  LIBSPDM_CRYPTO_ASN1_CONTEXT_SPECIFIC + 8);
+    if (result) {
```

Comment:
Why there is no check for vendorInfo and flags?

Reply:
Because there is no vendorInfo and flags reference.  The check will skip.

Reply:
What if the verifier would like to check vendorInfo and/or flags against references?

> Because there is no vendorInfo and flags reference. The check will skip.

Reply:
Please note, this is only reference implementation.

In a real production, the integrator can decide to add more fields to check, or skip some fields.